AsyncDisplayKit/ASDisplayNode+Subclasses.h

```diff
@@ -238,36 +238,58 @@ NS_ASSUME_NONNULL_BEGIN
 - (void)interfaceStateDidChange:(ASInterfaceState)newState fromState:(ASInterfaceState)oldState ASDISPLAYNODE_REQUIRES_SUPER;
 
 /**
- * @abstract Called whenever the visiblity of the node changed.
+ * @abstract Called whenever the node becomes visible.
  *
  * @discussion Subclasses may use this to monitor when they become visible.
+ *
+ * @note This method can be called from any thread and should therefore be thread safe.
  */
-- (void)visibilityDidChange:(BOOL)isVisible ASDISPLAYNODE_REQUIRES_SUPER;
+- (void)didEnterVisibleState ASDISPLAYNODE_REQUIRES_SUPER;
 
 /**
- * @abstract Called whenever the visiblity of the node changed.
+ * @abstract Called whenever the node is no longer visible.
  *
- * @discussion Subclasses may use this to monitor when they become visible.
+ * @discussion Subclasses may use this to monitor when they are no longer visible.
+ *
+ * @note This method can be called from any thread and should therefore be thread safe.
  */
-- (void)visibleStateDidChange:(BOOL)isVisible ASDISPLAYNODE_REQUIRES_SUPER;
+- (void)didExitVisibleState ASDISPLAYNODE_REQUIRES_SUPER;
 
 /**
- * @abstract Called whenever the the node has entered or exited the display state.
+ * @abstract Called whenever the the node has entered the display state.
  *
  * @discussion Subclasses may use this to monitor when a node should be rendering its content.
  *
  * @note This method can be called from any thread and should therefore be thread safe.
  */
-- (void)displayStateDidChange:(BOOL)inDisplayState ASDISPLAYNODE_REQUIRES_SUPER;
+- (void)didEnterDisplayState ASDISPLAYNODE_REQUIRES_SUPER;
 
 /**
- * @abstract Called whenever the the node has entered or left the load state.
+ * @abstract Called whenever the the node has exited the display state.
+ *
+ * @discussion Subclasses may use this to monitor when a node should no longer be rendering its content.
+ *
+ * @note This method can be called from any thread and should therefore be thread safe.
```

Comment:
My thought is we should guarantee that these will be called on main, but what does everyone else think about signing ourselves up for maintaining that?